data/transactions/logic/opcodes.go

```diff
@@ -61,38 +63,112 @@ const createdResourcesVersion = 6
 // experimental-
 const fidoVersion = 7 // base64, json, secp256r1
 
+type linearCost struct {
+	baseCost  int
+	chunkCost int
+	chunkSize int
+}
+
+func (lc *linearCost) compute(stack []stackValue) int {
+	cost := lc.baseCost
+	if lc.chunkCost != 0 && lc.chunkSize != 0 {
+		cost += lc.chunkCost * (len(stack[len(stack)-1].Bytes) + lc.chunkSize - 1) / lc.chunkSize
```

Comment:
I originally didn't intend to have a chunk, but in many cases the per byte size is < 1, so chunks allow you to express that: Every 8 byte chunk ands 3 cost units, for example.

As for why it's `(len + size - 1)/size` rather than `len/size`, it's because that essentially gives ceil(len/size) where normal integer division gives floor(lev/size).  I chose ceil because that happens to be the way that ethereum discretizing costs for hashing.  Hashing 0 bytes is one cost, but hashing anything from 1 byte to 32 bytes is the next cost.

Reply:
I would prefer a divceil() function to self document. I presume it would get inlined.

Reply:
For completeness, #3867 implements the feedback.  Resolving.